Common/Logging/DatabaseLogger.cs

```diff
@@ -30,8 +30,8 @@ protected override void SaveLog(LogEntry logEntry)
                 (
                     _connectionAddress,
                     ProcedureName, 
-                    new SqlParameter("@ApplicationName", SqlDbType.VarChar) {Value = logEntry.ApplicationName},
-                    new SqlParameter("@ApplicationVersion", SqlDbType.VarChar) {Value = logEntry.ApplicationVersion.ToString()},
+                    new SqlParameter("@ApplicationName", SqlDbType.VarChar) {Value = (object) logEntry.ApplicationName ?? DBNull.Value},
```

Comment:
seems like an odd approach to cast to an object where it will likely convert to a string under the covers. Just an observation that this might be done much cleaner.

Reply:
It won't be converted to a string underneath - `SqlParameter` is a typed native binding for SQL Server. This syntax is required to compile - `DBNull.Value` is of type `object` while `ApplicationName` is of type `string`, so it has to be downcast.